.github/workflows/build.yml

```diff
@@ -351,6 +351,28 @@ jobs:
       - name: Install cross
         uses: taiki-e/install-action@cross
       - run: cross build -p rustls --locked --target i686-unknown-linux-gnu
+      - run: cross test -p rustls --target i686-unknown-linux-gnu
+
+  cross-test-targets:
+    name: cross build-test - multiple targets
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        target:
+          - armv7-linux-androideabi
+          - s390x-unknown-linux-gnu
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+
+      - name: Install rust toolchain
+        uses: dtolnay/rust-toolchain@stable
+      - run: cargo install cross --git https://github.com/cross-rs/cross
```

Comment:
Why can't we use `taiki-e/install-action@cross` here?

Couldn't i686 be part of the target matrix here?

Reply:
> Couldn't i686 be part of the target matrix here?

Agreed - I think the bindgen installation bits could be conditional on the non-i686 target and then both jobs could be folded into one.

Reply:
Using that GitHub action did not work for me, kept getting CMake issues both with `taiki-e/install-action@cross` and now with this:

```yaml
      - name: Install cross
        uses: taiki-e/install-action@v2
        with:
          tool: cross
```

I will fold all of these jobs together, skipping the bindgen installation for the i686 (`i686-unknown-linux-gnu`) target.

Reply:
I would actually like to remove the existing `i686-unknown-linux-gnu` target in favor of `i686-linux-android`, which seems to be working in the CI.

Reply:
I'm not sure I understand `i686-unknown-linux-gnu` is cross-compiling fine in CI today, no?

Reply:
Alright I now have all the targets together, including `i686-unknown-linux-gnu`, multiple Android CPU build targets, and exotic `s390x-unknown-linux-gnu` target. I would love to add some more exotic & embedded targets, maybe in a separate PR (someday). There may be some overlap, should not be too bad at this point.